File: datastructuresInPython/LinkedList.py
class Node:
    next = None
    data = None
    def __init__(self,data):
        self.data = data

class LinkedList:

    head = None
    tail = None
    
    def __init__(self):
        ...

# Esse metodo de push q eu ainda não entendi
    def push(self,value):
        current = Node(value)
        if(self.head is None):
            self.head = current
            self.head.next = self.tail
        else:
            self.tail.next = current
        self.tail = current
#===========================================
    def pop(self):
        self.head = self.head.next

    def dequeue(self):
        current = self.head 
        while(current.next is not self.tail):
            current = current.next
        current.next = None
        self.tail = current

    def showLinkedList(self):
        current = self.head
        while(current is not None):
            print(current.data)
            current = current.next
        
    def showHeadAndTail(self):
        print("head: ",self.head.data,"\ntail: ",self.tail.data)    







if __name__ == "__main__":
    
    ll = LinkedList()
    
    ll.push(2)
    ll.push(34)
    ll.push(9)
    ll.push(8)
    ll.push(44)
    ll.push(4)
    ll.push(54)
    ll.dequeue()
    ll.showLinkedList()
    ll.showHeadAndTail()

File: LinkedList.cpp
#include <iostream>
#include <cstdio>

class LLNode{
public:
    int data;
    LLNode *next;

    LLNode(int value){
        this->data = value;
    }
};

class LinkedList{
private:
    LLNode* head;
    LLNode* tail;
    
public:
    LinkedList(){
        this->head = nullptr;
        this->tail = nullptr;
    }

    void push(int value){
        LLNode* newNode = new LLNode(value);
        if(this->head == nullptr){
            this->head = newNode;
            this->tail = newNode;
        }
        this->tail->next = newNode;
        this->tail = newNode;
        
    };

    void pop(){
        LLNode *current = this->head;
        while(current->next != this->tail){
            current = current->next;
        }
        current->next = nullptr;
        this->tail = current;
    };

    void show(){
        LLNode *current = this->head;
        while(current != this->tail){
            std::cout << current->data << "\n";
            current = current->next;
        }
        std::cout << current->data << "\n";
        std::cout << "\nhead: " << this->head->data;
        std::cout << "\ntail: " << this->tail->data << "\n";
    };

};